### PR TITLE
Add support for running fakeredis on 3.12

### PR DESCRIPTION
**Pull request by ChrisLovering**
Got [a PR](https://github.com/python-discord/sir-lancebot/pull/1486) up on lance ready to test this as needed.

### CODE REVIEW

File: pydis_core/utils/interactions.py [hunk 1]
```diff
@@ -23,9 +23,10 @@ def user_has_access(
         allowed_users: A sequence of user ids that are allowed access
         allowed_roles: A sequence of role ids that are allowed access
     """
-    if user.id in allowed_users or any(role.id in allowed_roles for role in getattr(user, "roles", [])):
-        return True
-    return False
+    return bool(
```
Comment (mbaruh) [inline pydis_core/utils/interactions.py:26]: why `bool(`?

File: docs/conf.py
Comment (mbaruh) [inline docs/conf.py:1]: What does this change mean?

Comment (ChrisLovering) [reply docs/conf.py:1]: When running `poetry run task docs` on Python 3.12 without this change, it errors saying that it couldn't import `docs`.

I noticed we were adding the project root to the path further down anyway so just moved it up.

File: pydis_core/utils/interactions.py (hunk 1 above)
Comment (ChrisLovering) [reply pydis_core/utils/interactions.py:26]: It's just what ruff auto-corrected it to. Just a bracket would probably work too, but I found this more readable.

**Review by wookie184 — APPROVED** [review]

**Pull request merged**